src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -50,6 +50,25 @@ auto run_cmd(mp::SSHSession& session, std::string&& cmd)
     return run_cmd(session, std::forward<std::string>(cmd), error_handler);
 }
 
+// Run a command returning a string which can end in space on a given session and reads the result. It adds a
+// non-space character (actually, a '-') to signal the end of the resulting string, and later removes the trailing
+// spaces and the special character to gather the contents of the return string including trailing spaces of the
+// command output.
+std::string run_string_cmd(mp::SSHSession& session, std::string&& cmd)
+{
+    std::string ret;
+    ret = run_cmd(session, "echo `" + cmd + "`-");
```

Comment:
One thing that irks me here is that if you try to `run_string_cmd` that has backticks (a subshell, I mean), this would break. At least, these should be `$( ... )`.

Reply:
Right